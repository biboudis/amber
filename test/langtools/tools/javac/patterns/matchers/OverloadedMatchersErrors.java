/*
 * @test /nodynamiccopyright/
 * @summary Verify error related to annotations and patterns
 * @enablePreview
 * @compile/fail/ref=OverloadedMatchersErrors.out -XDrawDiagnostics OverloadedMatchersErrors.java
 */
public class OverloadedMatchersErrors {
    private static int test(D o) {
        if (o instanceof D(String data, Integer out)) { // no compatible matcher found
            return out;
        }
        return -1;
    }

    private static int test2(D2 o) {
        if (o instanceof D2(String data, Integer out)) { // ambiguous
            return out;
        }
        return -1;
    }

    public record D() {
        public __matcher D(Object v1, Float out) {
            out = 10.0f;
        }

        public __matcher D(Float out, Integer v1) {
            out = 2;
        }
    }

    public record D2() {
        public __matcher D2(Object v, Integer out) {
            out = 1;
        }

        public __matcher D2(CharSequence v, Integer out) {
            out = 2;
        }
    }
}
