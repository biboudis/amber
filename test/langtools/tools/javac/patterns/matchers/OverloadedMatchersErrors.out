OverloadedMatchersErrors.java:9:26: compiler.err.no.compatible.matcher.found
OverloadedMatchersErrors.java:16:26: compiler.err.matcher.overloading.ambiguity
OverloadedMatchersErrors.java:28:19: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: int, java.lang.Float)
- compiler.note.preview.filename: OverloadedMatchersErrors.java, DEFAULT
- compiler.note.preview.recompile
3 errors